components/common/Tabs.tsx

```diff
@@ -0,0 +1,57 @@
+import { classNames } from '@/lib/util'
+
+export type TabView<ViewName> = {
+  view: ViewName
+  label: string
+  sublabel?: string
+}
+
+export function Tabs<ViewName>({
+  currentView,
+  onSetCurrentView,
+  views,
+}: {
+  currentView: ViewName
+  onSetCurrentView: (v: ViewName) => void
+  views: TabView<ViewName>[]
+}) {
+  const Tab = ({
+    view,
+    label,
+    sublabel,
+  }: {
```

Comment:
Defining a component anonymously like this has a quirk, which is that it always fully unmounts and remounts on render.  I'd recommend just pulling this out into its own component outside of `Tabs` render function.